### PR TITLE
side effects cache

### PR DESCRIPTION
**Pull request by etamponi**

### CODE REVIEW

**Comment by zkochan** [comment]
A test will be needed here. And in package-store a similar one.

**Comment by etamponi** [comment]
For the tests: I can't write them until we publish the new version of package-store, right? I mean, to write them now I would need to manually link the dependencies in `__package_preview__` and so on, but if we first release a new version of package-store, writing the tests here would be easier.

**Comment by zkochan** [comment]
sure, let's wait till other parts are published

File: test/index.ts [hunk 1]
```diff
@@ -75,6 +78,35 @@ test('server', async t => {
...
+
+  await storeCtrl.upload(path.join(__dirname, 'side-effect-fake-dir'), {
+    engine: fakeEngine,
+    pkgId: fakePkgId,
+  })
+
+  const cachePath = path.join('.store', fakePkgId, 'side_effects', fakeEngine, 'package')
+  t.ok(await fs.exists(cachePath), 'cache not uploaded')
+  t.ok(await fs.exists(path.join(cachePath, 'side-effect.js')), 'cache not uploaded: missing side-effect.js')
+  t.ok(await fs.exists(path.join(cachePath, 'side-effect.txt')), 'cache not uploaded: missing side-effect.txt')
```
Comment (zkochan) [inline test/index.ts]: the third parameter should describe what is tested. It is not a message that is printed on fail.

So it should be something like `side-effect.txt is uploaded`

Also, this test can be shorter

```js
t.deepEqual(await fs.readdir(cachePath), ['side-effect.js', 'side-effect.txt'], 'all files are uploaded')
```

**Comment by zkochan** [comment]
:ship: `0.7.0`

**Pull request merged**